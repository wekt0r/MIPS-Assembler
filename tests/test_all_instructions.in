lui $s1, 1
addi $s1, $s2, 1
addiu $s1, $s2, 1
slti $s1, $s2, 1
sltiu $s1, $s2, 1
andi $s1, $s2, 1
ori $s1, $s2, 1
xori $s1, $s2, 1
sll $s1, $s2, 1
srl $s1, $s2, 1
sra $s1, $s2, 1
sllv $s1, $s2, $s3
srlv $s1, $s2, $s3
srav $s1, $s2, $s3
mfhi $s1
mflo $s1
mult $s1, $s2
multu $s1, $s2
div $s1, $s2
divu $s1, $s2
add $s1, $s2, $s3
addu $s1, $s2, $s3
sub $s1, $s2, $s3
subu $s1, $s2, $s3
and $s1, $s2, $s3
or $s1, $s2, $s3
xor $s1, $s2, $s3
nor $s1, $s2, $s3
slt $s1, $s2, $s3
sltu $s1, $s2, $s3
