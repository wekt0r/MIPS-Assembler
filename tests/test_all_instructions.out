.text
00000000	3C110001	lui	$s1,1
00000004	22510001	addi	$s1,$s2,1
00000008	26510001	addiu	$s1,$s2,1
0000000c	2A510001	slti	$s1,$s2,1
00000010	2E510001	sltiu	$s1,$s2,1
00000014	32510001	andi	$s1,$s2,1
00000018	36510001	ori	$s1,$s2,1
0000001c	3A510001	xori	$s1,$s2,1
00000020	00128840	sll	$s1,$s2,1
00000024	00128842	srl	$s1,$s2,1
00000028	00128843	sra	$s1,$s2,1
0000002c	02728804	sllv	$s1,$s2,$s3
00000030	02728806	srlv	$s1,$s2,$s3
00000034	02728807	srav	$s1,$s2,$s3
00000038	00008810	mfhi	$s1
0000003c	00008812	mflo	$s1
00000040	02320018	mult	$s1,$s2
00000044	02320019	multu	$s1,$s2
00000048	0232001A	div	$s1,$s2
0000004c	0232001B	divu	$s1,$s2
00000050	02538820	add	$s1,$s2,$s3
00000054	02538821	addu	$s1,$s2,$s3
00000058	02538822	sub	$s1,$s2,$s3
0000005c	02538823	subu	$s1,$s2,$s3
00000060	02538824	and	$s1,$s2,$s3
00000064	02538825	or	$s1,$s2,$s3
00000068	02538826	xor	$s1,$s2,$s3
0000006c	02538827	nor	$s1,$s2,$s3
00000070	0253882A	slt	$s1,$s2,$s3
00000074	0253882B	sltu	$s1,$s2,$s3
